File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <iostream>
#include <stack>
using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    for(int i=0;i<moves.size();i++){
        moves[i]-=1;
    }
    stack<int> stack;
    int answer = 0;
    int col=board.size();
    for(int i=0;i<moves.size();i++){
        for(int j=0;j<col;j++){
            if(board[j][moves[i]]!=0){
                if(!stack.empty() && stack.top()==board[j][moves[i]]){
                  stack.pop();
                  answer+=2;
                }
                else{
                  stack.push(board[j][moves[i]]);
                }
                board[j][moves[i]]=0;
                break;
              }
                
            }
        }
  return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for(int i=0;i<moves.length;i++){
            moves[i]--;
        }
        for(int move: moves){
            for(int i=0;i<board[move].length;i++){
                if(board[i][move]!=0){
                    if(stack.empty()){
                        stack.push(board[i][move]);
                        board[i][move]=0;
                        break;
                    }
                    else{
                        if(board[i][move]==stack.peek()){
                            stack.pop();
                            board[i][move]=0;
                            answer+=2;
                            break;
                        }
                        else{
                            stack.push(board[i][move]);
                            board[i][move]=0;
                            break;
                        }
                    }
                }
            }
        }
        return answer;
    }
}